lib/updatehub/updatehub.c

```diff
@@ -146,6 +146,17 @@ is_compatible_hardware(struct resp_probe_some_boards *metadata_some_boards)
 	return false;
 }
 
+static void cleanup_connection(void)
+{
+	if (close(ctx.sock) < 0) {
+		LOG_ERR("Could not close the socket");
+	}
+
+	memset(&ctx.fds[1], 0, sizeof(ctx.fds[1]));
+	ctx.nfds = 0;
+	ctx.sock = 0;
+}
+
```

Comment:
Isn't this conflicting with #27223, do we need to have this change in this PR at all?

Reply:
The @27223 was already merged and it is safe now.